### PR TITLE
Paulo Barata

### PR DESCRIPTION
**Pull request by PBara7a**

### CODE REVIEW

File: src/components/breweryLi.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import addressInfo from "./addressInfo.js"
+import nameHeading from "./nameHeading.js"
+import phoneInfo from "./phoneInfo.js"
+import typeStamp from "./typeStamp.js"
+import websiteUrl from "./websiteUrl.js"
```
Comment (Creatorrr1) [inline src/components/breweryLi.js:5]: I Like the way you imported and exported code from one file to another. This is a great takeaway, I'll look into how I can do that.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import breweryLi from "./components/breweryLi.js";
+import breweries from "../templates/breweryObject.js";
+
+const breweriesList = document.getElementById('breweries-list')
+const brewery = breweries[0]
+
```
Comment (Creatorrr1) [inline src/index.js:11]: Nice putting an index[0] of array into a variable

File: src/render.js [hunk 3]
```diff
@@ -4,20 +4,21 @@ import state from "./state.js";
...
 
   const filterFlag = Boolean([...state.cityFilters].length)
   if (filterFlag) { // Render option if there are filters checked
   breweriesList.innerHTML = ''
     data.forEach(element => {
-      const { city } = element
-      if (state.cityFilters.has(city)) breweriesList.append(breweryLi(element))
+      const { city, brewery_type: type } = element
+      if (state.cityFilters.has(city) && targetBreweryTypes.includes(type)) {
+        breweriesList.append(breweryLi(element))
```
Comment (Creatorrr1) [inline src/render.js:15]: Amazing way of filtering

File: src/render.js [hunk 4]
```diff
@@ -4,20 +4,21 @@ import state from "./state.js";
...
     })
     return
   }
 
-  const targetBreweryTypes = ['micro', 'regional', 'brewpub']
   breweriesList.innerHTML = ''
-
   data.forEach(element => { // Normal rendering
     const { brewery_type: type } = element
     if (targetBreweryTypes.includes(type)) breweriesList.append(breweryLi(element))
```
Comment (Creatorrr1) [inline src/render.js:24]: Love how you created a condition and filtered what gets appended into the list.

File: src/state.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+const state = {
+  currentUrl: '',
+  typeExtension: '',
+  nameExtension: '',
+  cityFilters: new Set()
+}
+
+export default state
```
Comment (Alex90Jennings) [inline src/state.js:8]: nice way to set up the project with a state class

File: src/state.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const state = {
+  currentUrl: '',
```
Comment (Alex90Jennings) [inline src/state.js:2]: I never used the URL to filter the data, something I would like to try

File: src/fetchByName.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  const nameExtension = name ? `&by_name=${name}` : ''
+  fetch(`${state.currentUrl}${state.typeExtension}${nameExtension}`)
+  .then(res => res.json())
+  .then(data => {
+    state.nameExtension = nameExtension
+    render(data)
+  })
+}
+
+export default fetchByName
```
Comment (Alex90Jennings) [inline src/fetchByName.js:14]: these classes are well named and really easy to read

File: src/renderCityCheckBoxes.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import cityCheckbox from "./components/cityCheckbox.js";
+import cityLabel from "./components/cityLabel.js";
+
+function renderCityCheckBoxes(data, targetTypes) {
+  const citiesList = document.getElementById('filter-by-city-form')
+  citiesList.innerHTML = ''
+  const citiesSet = new Set()
```
Comment (Alex90Jennings) [inline src/renderCityCheckBoxes.js:7]: I wasn't able to get the cities filter to work, it seems like you have saved all the breweries from a specific city into a class and then rendered them from there?

File: src/index.js (hunk 2 above)
Comment (PBara7a) [reply src/index.js:11]: I was focusing on creating the template for the breweries Lis, so just needed one exemplar at this point.

File: src/renderCityCheckBoxes.js (hunk 8 above)
Comment (PBara7a) [reply src/renderCityCheckBoxes.js:7]: Never saved any breweries, I did requests to the Open Brewery DB with all necessary filters apart from the city ones and then worked with the data retrieved. The reason for this was that they do not offer a way to use multiple city filters, so I had to find another way to do it. My approach was to have a set storing all cities that had the filter checkbox checked and then render Lis only for those breweries from cities existing in the set.

**Pull request closed without merging**